src/sentry/snuba/discover.py

```diff
@@ -360,13 +344,16 @@ def timeseries_query(
         if comparison_delta:
             if len(base_builder.aggregates) != 1:
                 raise InvalidSearchQuery("Only one column can be selected for comparison queries")
-            comp_query_params = deepcopy(params)
-            comp_query_params["start"] -= comparison_delta
-            comp_query_params["end"] -= comparison_delta
+            comp_query_params = snuba_params.copy()
+            assert comp_query_params.start is not None, "start is required"
+            assert comp_query_params.end is not None, "end is required"
+            comp_query_params.start -= comparison_delta
+            comp_query_params.end -= comparison_delta
```

Comment:
I sorta hate this block 🤔 after the follow up PR to remove params entirely might add a utility to clean this up